### PR TITLE
[WIP] Redesign settings pages

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Implements new layouts for settings pages, and implements those layouts on the following pages:

1. InitialSettingsPage
1. WorkspacesListPage
1. PreferencesPage
1. SecuritySettingsPage
1. AboutPage

### Fixed Issues
$ https://github.com/Expensify/App/issues/12261

### Tests
1. Sign into an account without any workspaces.
1. Open the settings page. Verify it looks good.
1. Open the Preferences page. Verify that the animation runs and the page looks good.
1. Open the security settings page. Verify that the animations runs and the page looks good.
1. Open the About page. Verify that the animation runs and the page looks good.
1. Open the Workspace list page. Verify that the animation runs and the page looks good.
1. Create a workspace. Verify that the page still looks good.

- [x] Verify that no errors appear in the JS console

### Offline tests
n/a

### QA Steps
Same as tests, just use your judgement and look for any layout issues or bugs.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

image

image

image

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/47436092/f05d3310-9e70-4f42-ab54-b4519e551866

</details>

<details>
<summary>Desktop</summary>

image

image

image

image

image

https://github.com/Expensify/App/assets/47436092/6e922304-fadb-460c-9b6b-8ef7ba365bfc

</details>

<details>
<summary>iOS</summary>

image

image

image

https://github.com/Expensify/App/assets/47436092/7bded672-5d31-4e46-be43-5599e46be03e

</details>

<details>
<summary>Android</summary>

_w/ updated animations:_

https://github.com/Expensify/App/assets/47436092/5f363dd0-e693-418d-a592-ce0e9f747d49

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
Woo! Any screenies available to preview?

**Comment by roryabraham** [comment]
> Woo! Any screenies available to preview?

Not yet, but I did manage to resolve [this issue](https://github.com/Expensify/App/issues/13001) with [this PR](https://github.com/Expensify/App/pull/15471). Like I said, that was the biggest issue I was having. So hopefully we can get that merged soon and get this unblocked. It will also substantially simplify this PR.

**Comment by shawnborton** [comment]
Sounds great! Keep me posted when there are some early screenshots to review please, thank you!

**Comment by roryabraham** [comment]
@shawnborton ready for review

**Comment by shawnborton** [comment]
Comparison:

PR | Mockup
----- | -----
image | image

**Comment by shawnborton** [comment]
Maybe it's mostly mobile web chrome that looks the most off?

**Comment by roryabraham** [comment]
Also @shawnborton I think that the lottie animation for the security page is kind of off-center, idk if you noticed the same.

**Comment by shawnborton** [comment]
I agree with you there!

I was hoping to get this as close to its final form as possible, and then we can take videos of the various pages and share with Augenblick to get some adjustments. If you have a good video of the vault for instance, I can share with them so they can move some negative space around.

**Comment by roryabraham** [comment]
Just noting here that I've observed some frame-drops on iOS safari for the Security and About animations. For some reason, the Workspaces and Preference animations are running much smoother.

**Comment by roryabraham** [comment]
There's also this minor bug on iOS safari where the safe is layered wrong or something:

**Comment by roryabraham** [comment]
@shawnborton I've made changed according to your feedback. It looks better + more consistent now, and the styles have been substantially simplified. I updated all the screenshots and included videos for some platforms as well so you can see the animations in action.

**Comment by shawnborton** [comment]
Cool, started some conversation in the TCW room about this, we'll see what Augenblick says.

**Comment by roryabraham** [comment]
I figured out an easy way to squeeze better performance out of these animations on web + to fix the weird display bug:

### Before (svg renderer)
https://github.com/Expensify/App/assets/47436092/b6422bd4-687b-4b09-9017-0b39f7c7a14f

### After (canvas renderer)
https://github.com/Expensify/App/assets/47436092/a814a867-2e34-4857-adc7-75b9ac408a41

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/868c329c-7b77-4ad2-b3d0-299ff801d4f1


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/25876548/12566552-1ade-4947-8f9f-87304fe9b524


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/ca203925-9ea2-4606-972e-f3a399cb0601




</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/79f4fd06-f339-41b4-970c-11823272c42d



</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25876548/ea9ca899-5c1b-4284-abc0-1a474fe06d48


</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/25876548/089bc2ac-3b4d-4905-8dc0-593f54d291f4


</details>

**Comment by shawnborton** [comment]
How is this screen happening here? 
image

**Comment by sobitneupane** [comment]
Everything is working great. Animations are working well. Few Questions:

- "You appear to be offline" has background color same as the status bar. Is it expected?
Screenshot 2023-05-12 at 17 00 46

- Workspace General Settings page has purple background around the animation in Android/Native (tested in emulator).
Screenshot 2023-05-12 at 16 38 33

- Transition of status bar is quite slow. Is it expected?

https://github.com/Expensify/App/assets/25876548/06dee933-e56e-420a-a6b6-61a9ed63f3b5

File: src/pages/settings/Preferences/PreferencesPage.js [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 import * as User from '../../../libs/actions/User';
-import ScreenWrapper from '../../../components/ScreenWrapper';
 import Switch from '../../../components/Switch';
 import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
 import compose from '../../../libs/compose';
 import withEnvironment, {environmentPropTypes} from '../../../components/withEnvironment';
 import TestToolMenu from '../../../components/TestToolMenu';
 import MenuItemWithTopDescription from '../../../components/MenuItemWithTopDescription';
+import IllustratedHeaderPageLayout from '../../../components/IllustratedHeaderPageLayout';
+import PreferencesDJAnimation from '../../../../assets/animations/PreferencesDJ.json';
```
Comment (grgia) [inline src/pages/settings/Preferences/PreferencesPage.js:22]: Should we be importing these directly or port them through a file similar to how we handle illustrations/icons with Expensicons? That way if we reuse animations and need to update the file name or something we could do it in a single file

File: src/pages/workspace/WorkspacesListPage.js [hunk 2]
```diff
@@ -178,32 +180,60 @@ class WorkspacesListPage extends Component {
...
+                                    <Icon
+                                        src={icon}
+                                        width={variables.iconSizeSuperLarge}
+                                        height={variables.iconSizeSuperLarge}
+                                    />
+                                    <Text style={[styles.h3, styles.pl2]}>{this.props.translate(translationKey)}</Text>
+                                </View>
+                            ))}
+                        </View>
+                    </View>
```
Comment (grgia) [inline src/pages/workspace/WorkspacesListPage.js:233]: Should we make these feature lists their own component? I think the illustration + feature pattern is likely to be repeated. Or should we save that refactor until it is repeated?

**Comment by roryabraham** [comment]
Ok, so unfortunately this PR was completely blown up by the react-navigation refactor. So I'm going to try and take it more piece-by-piece and break it up into a few smaller PRs.

**Comment by roryabraham** [comment]
Marking this a WIP as I break it up into more managable pieces and ultimately resolve all the conflicts from the react-navigation refactor.

**Comment by shawnborton** [comment]
Sounds good. Do you have a rough ETA of when you might be able to finish this one by?

**Comment by grgia** [comment]
any reason to keep this PR open? @roryabraham

**Comment by roryabraham** [comment]
I have been referencing it but we can close it. @grgia feel free to review the latest 
here: https://github.com/Expensify/App/pull/21852

**Pull request closed without merging**